src/els_compiler_diagnostics.erl

```diff
@@ -48,6 +48,8 @@ diagnostics(Uri) ->
       %% file, since that allows us to identify most of the common
       %% errors in header files.
       parse(Uri);
+    <<".escript">> ->
```

Comment:
See my comment in the PR. The alternative would be to treat "all" files as potential escripts, but that would cause lot of un-necessary work.

Reply:
We actually have some files ending `.erl` that have the escript shebang.

Reply:
I can imagine you're not the only one, but let's take baby steps :)
Is there an actual convention on the extension? Checking...

Reply:
Actually:

```
escript provides support for running short Erlang programs without having to compile them first, and an easy way to retrieve the command-line arguments.

It is possible to bundle escript(s) with an Erlang runtime system to make it self-sufficient and relocatable. In such a standalone system, the escript(s) should be located in the top bin directory of the standalone system and given .escript as file extension. Further the (built-in) escript program should be copied to the same directory and given the scripts original name (without the .escript extension). This will enable use of the bundled Erlang runtime system.
```

So, I would say `.escript` is the correct extension. Maybe you should rename those? :)

Reply:
I guess people would use `.erl` as the extension mainly to get syntax highlighting, but you just fixed that in VSCode...